src/core/server/legacy/plugins/find_legacy_plugin_specs.ts

```diff
@@ -39,7 +39,7 @@ export interface LegacyPluginSpec {
 }
 
 export async function findLegacyPluginSpecs(settings: unknown, loggerFactory: LoggerFactory) {
-  const configToMutate: Config = defaultConfig(settings);
+  const configToMutate: LegacyConfig = defaultConfig(settings);
```

Comment:
So this was actually incorrectly typed. This is not a NP Config, but the legacy `KibanaConfig` config. (Had an issue because of this in the validation code and took me quite some time to find where the root cause was) Had to adapt quite a lot of types in legacy and savedObject code, as this config is propagated after discover in `LegacyServiceDiscoverPlugins` and used in savedObject service. Also had to create the `LegacyConfig` type in NP as we did not have a proper representation for it.